app/models/service_template_transformation_plan.rb

```diff
@@ -73,11 +78,26 @@ def self.validate_config_info(options)
 
     raise _('Must provide an existing transformation mapping') if mapping.blank?
 
-    vms = if config_info[:vm_ids]
-            VmOrTemplate.find(config_info[:vm_ids])
-          else
-            config_info[:vms]
-          end
+    pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
+    post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
+
+    vms = []
+    if config_info[:actions]
+      vm_objects = VmOrTemplate.where(:id => config_info[:actions].collect { |vm_hash| vm_hash[:vm_id] }.compact)
```

Comment:
You can make `vm_objects` a hash rather than an array. Then in Ln 89 you can directly lookup the object without a looping.
Ln 88 could be as simple as `vm_obj = vm_objects[vm_hash[:vm_id]] || vm_hash[:vm]`